### PR TITLE
Fix name transformation for ResourceData

### PR DESCRIPTION
**Pull request by live1206**
For while-listed single words, we are prepending the service name to distinguish them.
For models with suffix `Resource`, we should rename them with suffix `ResourceData`.

### CODE REVIEW

File: test/TestProjects/MgmtTest-TypeSpec/PrivateEndpointConnection.tsp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.ResourceManager;
+
+namespace Microsoft.MgmtTest;
+
+model PrivateEndpointConnectionResource
```
Comment (pshao25) [inline test/TestProjects/MgmtTest-TypeSpec/PrivateEndpointConnection.tsp:7]: Isn't one model and one operation enough?

File: test/TestProjects/MgmtTest-TypeSpec/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Release History
```
Comment (pshao25) [inline test/TestProjects/MgmtTest-TypeSpec/CHANGELOG.md:1]: Try namespace not start with "Microsoft" or "Azure", that might not generate this file.

File: test/TestProjects/MgmtTest-TypeSpec/PrivateEndpointConnection.tsp (hunk 1 above)
Comment (live1206) [reply test/TestProjects/MgmtTest-TypeSpec/PrivateEndpointConnection.tsp]: It is. Removed the other operations

File: test/TestProjects/MgmtTest-TypeSpec/CHANGELOG.md (hunk 2 above)
Comment (live1206) [reply test/TestProjects/MgmtTest-TypeSpec/CHANGELOG.md]: Updated

File: test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtResourceNameTests.cs [hunk 3]
```diff
@@ -59,6 +59,7 @@ public void ValidateResources(string resource, bool isExists)
         [TestCase("DisplayData", false)]
         public void ValidateResourceData(string resourceData, bool isExists)
         {
+            var all = FindAllResourceData().ToList();
```
Comment (pshao25) [inline test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtResourceNameTests.cs:62]: Do we need this line?

**Review by pshao25 — APPROVED** [review]

File: test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtResourceNameTests.cs (hunk 3 above)
Comment (live1206) [reply test/AutoRest.TestServer.Tests/Mgmt/TestProjects/MgmtResourceNameTests.cs]: reverted

**Pull request merged**